### PR TITLE
Wiz version 1.3.2

### PR DESCRIPTION
**Pull request by yanivblumWiz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- New command: wiz-copy-to-forensics-account
- Add evidence (detections) for TDR issues as well
- Add issue_type and fix entity_type filter name for wiz-get-issues command
- Deprecated wiz-rescan-machine-disk command

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by adi88d** [comment]
Hello @yanivblumWiz 
please fill the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)
and let me know when this pull request is ready for review
Thanks

File: Packs/Wiz/Integrations/Wiz/README.md [hunk 1]
```diff
@@ -67,7 +68,8 @@ Get the issues on cloud resources
 
 #### Command Example
 ```
-!wiz-get-issues issue_type="VIRTUAL_MACHINE"
+!wiz-get-issues entity_type="VIRTUAL_MACHINE"
+!wiz-get-issues issues_type="THREAT_DETECTION"
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/README.md:72]: issue_type (not issueS_type)

File: Packs/Wiz/Integrations/Wiz/README.md [hunk 2]
```diff
@@ -346,3 +329,23 @@ Clear a due date for an Issue.
...
+<h4> Input </h4>
+
+| **Argument Name** | **Description** | **Required** |
+| --- |-----------------| --- |
+| resource_id | Resource Id     | Required | 
+
+#### Command Example
+```
+!wiz-copy-to-forensics-account resource_id="12345678-1234-1234-1234-cc0a24716e0b"
+!wiz-copy-to-forensics-account resource_id="arn:aws:ec2:us-east-1:452225563321:instance/i-05r662bfb9708a4e8"
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/README.md:354]: Change the account to a fake number

File: Packs/Wiz/Integrations/Wiz/Wiz.py [hunk 3]
```diff
@@ -1370,6 +2358,53 @@ def get_project_team(project_name):
...
+            }
+        }
+        resource_id_response = checkAPIerrors(PULL_RESOURCES_ID_NATIVE_QUERY, variables)
+        if resource_id_response['data'] is None or not resource_id_response['data']['cloudResources']['nodes']:
+            demisto.error(f"Resource with ID {resource_id} not found.")
+            return f"Resource with ID {resource_id} not found."
+        else:
+            resource_id = resource_id_response['data']['cloudResources']['nodes'][0]['id']
+
+    copy_to_forensics_account_mutation = """
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:2382]: you don't want to move this mutation on the top of the script with all the queries/mutations?

File: Packs/Wiz/Integrations/Wiz/Wiz.py [hunk 4]
```diff
@@ -1157,6 +2226,14 @@ def get_issue_evidence(issue_id):
     # Getting the Issue Evidence Query
     issue_object = _get_issue(issue_id, is_evidence=True)
 
+    issue_type = issue_object['data']['issues']['nodes'][0]['type']
+
+    if issue_type == 'THREAT_DETECTION':
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:2231]: what if this is another type?

File: Packs/Wiz/Integrations/Wiz/Wiz.py [hunk 5]
```diff
@@ -1445,115 +2482,106 @@ def main():
...
                 reopen_note=reopen_note
             )
-            command_result = CommandResults(readable_output=issue_response, raw_response=issue_response)
+            command_result = CommandResults(readable_output=copy_mutation_response, raw_response=copy_mutation_response)
             return_results(command_result)
 
         elif command == 'wiz-resolve-issue':
             demisto_args = demisto.args()
-            issue_id = demisto_args.get(WizInputParam.ISSUE_ID)
+            resource_id = demisto_args.get(WizInputParam.ISSUE_ID)
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:2509]: why you don't name this param issue_id? (in all the other functions as well)

File: Packs/Wiz/Integrations/Wiz/Wiz.yml [hunk 6]
```diff
@@ -148,6 +148,13 @@ script:
       - LOW
       - INFORMATIONAL
       description: Get Issues of a specific severity and above.
+    - name: issue_type
+      auto: PREDEFINED
+      predefined:
+      - TOXIC_COMBINATION
+      - THREAT_DETECTION
+      - CLOUD_CONFIGURATION
+      description: Get Issues of a specific type.
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/Wiz.yml:157]: I think it should be 
- description (add the "-")

File: Packs/Wiz/Integrations/Wiz/command_examples.txt [hunk 7]
```diff
@@ -18,12 +19,13 @@
 
 !wiz-get-issue-evidence issue_id="12345678-1234-1234-1234-cc0a24716e0b"
 
-!wiz-rescan-machine-disk vm_id="arn:aws:ec2:us-east-2:123456789098:instance/i-1234abcd123456789"
-
 !wiz-set-issue-due-date issue_id="12345678-1234-1234-1234-cc0a24716e0b" due_at="2022-01-20"
 
 !wiz-clear-issue-due-date issue_id="12345678-1234-1234-1234-cc0a24716e0b"
+!wiz-copy-to-forensics-account resource_id="arn:aws:ec2:us-east-1:452225563321:instance/i-05r662bfb9708a4e8"
```
Comment (ariel-wiz) [inline Packs/Wiz/Integrations/Wiz/command_examples.txt:25]: also change the account id

File: Packs/Wiz/Integrations/Wiz/Wiz.py (hunk 4 above)
Comment (yanivblumWiz) [reply Packs/Wiz/Integrations/Wiz/Wiz.py:2231]: Then you proceed with the original flow that we had until now, which still works

File: Packs/Wiz/Integrations/Wiz/Wiz.yml (hunk 6 above)
Comment (yanivblumWiz) [reply Packs/Wiz/Integrations/Wiz/Wiz.yml:157]: I think the YAML is correct this way, I will check again and change if needed

File: Packs/Wiz/Integrations/Wiz/Wiz.py (hunk 5 above)
Comment (yanivblumWiz) [reply Packs/Wiz/Integrations/Wiz/Wiz.py:2509]: refactor mistake, reverting

**Comment by yanivblumWiz** [comment]
> Hello @yanivblumWiz please fill the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA) and let me know when this pull request is ready for review Thanks

Hi @adi88d 
The PR is ready for your review :)

**Review by adi88d — CHANGES_REQUESTED** [review]
Hello @yanivblumWiz,
Thank you for your contribution! the code looks great!
see my other comments below

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR. 

In the demo we want to test the commands that effected by the changes in `PULL_ISSUE_WITH_EVIDENCE_PARAM_QUERY`:
- `wiz-resolve-issue`
- `wiz-clear-issue-note`
- `wiz-get-issue-evidence` - when the issue is not exists and for issue type `THREAT_DETECTION`(with and without evidence).

- The new command: `wiz-copy-to-forensics-account` - test with invalid uuid, test when resource_id is not found.
- `wiz-get-resource` - test with resource_id and resource_name

Thanks again

File: Packs/Wiz/Integrations/Wiz/Wiz.py [hunk 8]
```diff
@@ -1370,6 +2429,46 @@ def get_project_team(project_name):
...
 
 
+def copy_to_forensics_account(resource_id):
+    """
+    Copy resource Volumes to a Forensics Account
+    """
+    demisto.info(f"resource id is {resource_id}\n")
+    demisto.debug("copy_to_forensics_account, enter")
+
+    if not is_valid_uuid(resource_id):
```
[the remark this answers is not in the document]
Comment (yanivblumWiz) [reply Packs/Wiz/Integrations/Wiz/Wiz.py:2439]: No, if the user inserted a non UUID, I am treating this as a provider ID (ARN/ Azure's ID etc...)

**Comment by yanivblumWiz** [comment]
> Hello @yanivblumWiz, Thank you for your contribution! the code looks great! see my other comments below
> 
> We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.
> 
> In the demo we want to test the commands that effected by the changes in `PULL_ISSUE_WITH_EVIDENCE_PARAM_QUERY`:
> 
> * `wiz-resolve-issue`
> * `wiz-clear-issue-note`
> * `wiz-get-issue-evidence` - when the issue is not exists and for issue type `THREAT_DETECTION`(with and without evidence).
> * The new command: `wiz-copy-to-forensics-account` - test with invalid uuid, test when resource_id is not found.
> * `wiz-get-resource` - test with resource_id and resource_name
> 
> Thanks again

I pushed a commit with all the fixes and responded where needed.
Regarding the demo, `wiz-resolve-issue` and `wiz-clear-issue-note` were not changed as they use the `PULL_ISSUES_QUERY` (evidence=False) on those commands.

**Comment by adi88d** [comment]
@yanivblumWiz please send me your email address

**Comment by yanivblumWiz** [comment]
> @yanivblumWiz please send me your email address

yaniv.blum@wiz.io

**Comment by adi88d** [comment]
@yanivblumWiz 
We had a Great demo today!
further to the demo, few comments:
- In `wiz-copy-to-forensics-account` command - please add to the CommandResults the outputs.
- `wiz-get-issue-evidence` - add to the CommandResults the outputs.
- `wiz-get-resource` - update the release notes with the changes in this command.
- `wiz-get-issues` - add limit argument

**Comment by yanivblumWiz** [comment]
Hi @adi88d thanks for the demo today!
As discussed, I updated the RN as you requested, and I will forward all the comments to our product team for the next version.
Let me know if we are missing anything to merge this PR

**Review by adi88d — APPROVED** [review]

**Pull request merged**